### PR TITLE
Add squid package

### PR DESCRIPTION
**Pull request by Zopolis4**
https://www.squid-cache.org/

### Run the following to get this pull request's changes locally for testing.
```
CREW_REPO=https://github.com/Zopolis4/chromebrew.git CREW_BRANCH=coilingtendrils crew update
```

### CODE REVIEW

File: packages/squid.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+require 'buildsystems/autotools'
+
+class Squid < Autotools
+  description 'Squid Web Proxy Cache'
+  homepage 'https://github.com/squid-cache/squid'
```
Comment (uberhacker) [inline packages/squid.rb:5]: Better home page? https://www.squid-cache.org

File: packages/squid.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  version '6.8'
+  license 'GPL-2.0'
+  compatibility 'all'
+  source_url 'https://www.squid-cache.org/Versions/v6/squid-6.8.tar.xz'
```
Comment (uberhacker) [inline packages/squid.rb:9]:
```
source_url 'https://github.com/squid-cache/squid.git'
git_hashtag 'SQUID_6_8'
```
?

File: packages/squid.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  depends_on 'libxml2' # R
+  depends_on 'linux_pam' # R
+  depends_on 'nettle' # R
+  depends_on 'openldap' # R
+  depends_on 'glibc' # R
+
+  # https://github.com/squid-cache/squid/pull/1118
+  pre_configure_options "CXXFLAGS='-Wno-alloc-size-larger-than'"
+  configure_options '--disable-arch-native --without-systemd'
+end
```
Comment (uberhacker) [inline packages/squid.rb:39]: Add to `tools/packages.yaml`?

**Comment by uberhacker** [comment]
Thank you for the new package @Zopolis4!  See comments for suggestions.

File: packages/squid.rb (hunk 2 above)
Comment (Zopolis4) [reply packages/squid.rb:9]: It's faster to download tarfiles than to download from git

Comment (uberhacker) [reply packages/squid.rb:9]: A git clone is a one-time build from source and you don't need to calculate the sha256sum hash.

Comment (Zopolis4) [reply packages/squid.rb]: My apologies, I timed both of them and downloading from git is indeed quite a bit faster.

Comment (uberhacker) [reply packages/squid.rb]: Good to know but there are certainly times where it is slower to clone the repo.

**Comment by satmandu** [comment]
(script....)

**Comment by Zopolis4** [comment]
I'm not sure-- at any rate, is it not possible for us to at least run an init manager as user for basic startup scripts? Rolling them ourselves seems unnecessary in the majority of cases (bar `sommelier`, where there is no upstream script)

**Comment by satmandu** [comment]
> I'm not sure-- at any rate, is it not possible for us to at least run an init manager as user for basic startup scripts? Rolling them ourselves seems unnecessary in the majority of cases (bar `sommelier`, where there is no upstream script)

It's a huge pain because the first crosh process we open isn't PID 1, which systemd needs.

**Comment by Zopolis4** [comment]
Oh, I was thinking of using a systemd alternative-- I think there are a few which can handle the service management and also don't require PID 1.

**Review by satmandu — APPROVED** [review]

**Pull request merged**